Add Compass Less operation page

Hwurzburg opened the pull request.

tridge commented on common/source/docs/common-compassless.rst:
```diff
@@ -0,0 +1,37 @@
+
+On Plane, a steady GPS heading is used after takeoff to align the /attitude/velocity/position/heading estimation filters (EKFs) for yaw. This usually occurs a few seconds after takeoff on Planes without compass. However, this approach cannot be used for VTOL navigation, since it needs to be accurate very shortly after takeoffs with no GPS velocity or heading inputs.
+
+Compass-less VTOL operation has been made possible by the addition of a Gaussian Sum Filter (GSF) which takes IMU and GPS inputs to compute a probable heading,if a yaw source (like a compass) is unavailable, or unhealthy.
```
missing space after comma

tridge commented on common/source/docs/common-compassless.rst:
```diff
@@ -0,0 +1,37 @@
+
+Compass-less VTOL operation has been made possible by the addition of a Gaussian Sum Filter (GSF) which takes IMU and GPS inputs to compute a probable heading,if a yaw source (like a compass) is unavailable, or unhealthy.
+
+.. note:: So far, the accuracy is quite acceptable, but this is still considered experimental and more experience and test flights by the community are required.
```
I think change "So far" to "In testing so far"

tridge commented on common/source/docs/common-compassless.rst:
```diff
@@ -0,0 +1,37 @@
+
+Simply, set :ref:`COMPASS_ENABLE<COMPASS_ENABLE>` = 0. Fallback to the GSF will be automatic if normal default values are being used for EKF parameters.
+
+.. note:: Compass-less operation depends on good velocity reports from the GPS. Do not attempt with older generation GPS, use at least UBlox MN8 or equivalent performance GPS. Also, be sure GPS view of the sky is unobstructed throughout the flight. Otherwise, poor heading estimation will result.
```
change MN8 to M8

rmackay9 commented on common/source/docs/common-compassless.rst:
```diff
@@ -0,0 +1,37 @@
+
+Afterwards, flight and navigation should be acceptable.
+
+If any problems are experienced, please post logs either on the https://discuss.ardupilot.org or via a GitHub issue at https://github.com/ArduPilot/ardupilot/issues.
```
I don't think we should recommend users directly create issues, best to just stick with the discuss option.

Hwurzburg replied:
done

rmackay9 commented on common/source/docs/common-compassless.rst:
```diff
@@ -0,0 +1,37 @@
+Operation without a Compass
+===========================
+
+Operation without a compass has been possible for Plane for a long time. However, recent additions of a compass/yaw source fallback mechanism in ArduPilot has opened the possibility of operations of  VTOL vehicles, such as Copters and QuadPlanes, without the use of a compass.
```
I think "recent additions of a compass/yaw source fallback mechanism in ArduPilot" could be changed to "ArduPilot 4.1 (and higher) includes a Gaussian Sum Filter (GSF) which may allow Copter, Rovers and QuadPlanes to fly without a compass"

rmackay9 commented:
Maybe we should add a link to my video: https://www.youtube.com/watch?v=3xW9hj-lxNU.  We've got some other feedback above but after these are addressed I think we should merge this.  Perhaps we should also add a note near the top of the page that GSF is only available in ArduPilot 4.1 (and higher)

Hwurzburg commented:
no need for notes on rev (I am against adding things that obsolete themselves over time, ie when firmware is beyond the note)....will add this to the Upcoming features page and it can be moved into mainstream wiki when it appears in a beta release...

Hwurzburg commented:
> Maybe we should add a link to my video: https://www.youtube.com/watch?v=3xW9hj-lxNU. We've got some other feedback above but after these are addressed I think we should merge this. Perhaps we should also add a note near the top of the page that GSF is only available in ArduPilot 4.1 (and higher)

added video, corrected, moved links to Upcoming so no version tag is required....merging...thanks!

AndKe commented:
Please add information that gives answers to this:
https://discuss.ardupilot.org/t/compassless-flight-plus-gsf-more-questions-than-answers/63732

As-is, it is hard to fully understand if GSF is supposed to be an aid to enable compassless flight when prepared for it, or as handling of compass error after takeoff,  or a fallback for all stages of flight.
